src/llama-grammar.cpp

```diff
@@ -682,6 +682,101 @@ static bool llama_grammar_match_partial_char(
     return !is_positive_char;
 }
 
+// transforms a grammar pushdown stack into N possible stacks, all ending
+// at a character range (terminal element)
+// additionally memorizes the stack to its possible stacks by mapping
```

Comment:
```suggestion
// additionally memoizes the stack to its possible stacks by mapping
```

Even though my spellchecker doesn't like it, "memoizes" and "memoizing" are proper forms of "[memoization](https://en.wikipedia.org/wiki/Memoization)".